### PR TITLE
CBO-783: Rerelease of Advocates view accessibility updates

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Investigate and fix accessibility issues in all views with routes matching 'advocates'.

#### Ticket
[CBO-783](https://dsdmoj.atlassian.net/browse/CBO-783)

#### Why
To make this section of CCCD accessible for all users. This was previously released, however reverted due to user identified issues. The issues have now replicated locally, tested and resolved.

#### How
- Add relevant tabindex
- Add semantically correct file upload, improve accessibility
- Remove redundant Dropzone partial and helper
- Proper usage of aria tag
- Properly defined table attributes

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -1416,19 +1416,21 @@ en:
...
+          action: 'Action'
+          status: 'Status'
           choose_file: 'Choose a file'
           file_name: "File name"
           supporting_evidence_docs: 'Upload supporting evidence'
           supporting_evidence: Supporting evidence
           file_upload_message: Drag and drop files here
           legend: Upload supporting evidence
           or: or
+          uploadeding_docs: 'Documents waiting to be uploaded'
```
Comment (jsugarman) [inline config/locales/en.yml]: **no-blocker** typo? uploadeding_docs -> uploaded_docs

File: app/views/external_users/claims/supporting_documents/_new.html.haml [hunk 2]
```diff
@@ -10,18 +10,23 @@
...
-          %br
-          .button-grey-3
-            = t('.choose_file')
+    .dropzone
+      = f.label :documents, 'Upload file', class: 'form-group'
+      = f.file_field :documents, name: 'documents', multiple: true, class: 'form-control'
+
+%table#dropzone-files.files.hidden
+  %caption
+    = t('.uploadeding_docs')
```
Comment (jsugarman) [inline app/views/external_users/claims/supporting_documents/_new.html.haml]: **no-blocker** typo? uploadeding_docs -> uploaded_docs

**Review by jsugarman — APPROVED** [review]
👍 LGTM. just one non-blocking typo you might want to change

File: app/assets/javascripts/modules/external_users/claims/Dropzone.js [hunk 3]
```diff
@@ -1,85 +1,200 @@
...
+    this.target = $('.dropzone');
 
     Dropzone.autoDiscover = false;
 
-    this.$target.dropzone({
+    if (self.dragAndDropSupported() && self.formDataSupported() && self.fileApiSupported()) {
+      this.target.addClass('dropzone-enhanced');
+      self.setupDropzone();
+      self.setupFileInput();
+      self.setupStatusBox();
```
Comment (willmcb) [inline app/assets/javascripts/modules/external_users/claims/Dropzone.js]: If `this` and `self` refer to the same thing, as per line 3, why use two different calls?

Comment (willmcb) [inline app/assets/javascripts/modules/external_users/claims/Dropzone.js:13]: If `this` and `self` refer to the same thing, as per line 3, why use two different calls?

**Pull request merged**